src/main/java/graphql/parser/StringValueParsing.java

```diff
@@ -135,15 +140,37 @@ public static String parseSingleQuotedString(String string) {
                     writer.write('\t');
                     continue;
                 case 'u':
-                    String hexStr = string.substring(i + 1, i + 5);
-                    int codepoint = Integer.parseInt(hexStr, 16);
-                    i += 4;
-                    writer.write(codepoint);
+                    i = parseEscapedUnicode(writer, string, i);
                     continue;
                 default:
                     Assert.assertShouldNeverHappen();
             }
         }
         return writer.toString();
     }
+
+    private static int parseEscapedUnicode(StringWriter writer, String string, int i) {
```

Comment:
I feel strongly that this should go into a Util class (UnicodeUtil say) and be unit tested on its own

Better units of code